samcli/lib/build/app_builder.py

```diff
@@ -960,6 +961,10 @@ def _build_function_on_container(
             # "/." is a Docker thing that instructions the copy command to download contents of the folder only
             result_dir_in_container = response["result"]["artifacts_dir"] + "/."
             container.copy(result_dir_in_container, artifacts_dir)
+
+        except (DockerImagePullFailedException) as ex:
+            raise BuildInsideContainerError(ex)
```

Comment:
nit: Should we adjust any wording inside the exception or relaying is OK (not sure what will be the output here)?

Reply:
Relay should be fine IMO.

```
Error: Could not find sam/custom-graal-iage image locally and failed to pull it from docker.
```

its pretty clear.